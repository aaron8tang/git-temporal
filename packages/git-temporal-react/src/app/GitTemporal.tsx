import React, { Component } from 'react';
import { connect } from 'react-redux';

import { GitTemporalProps, DispatchProps, StateProps } from 'app/interfaces';
import { selectPath } from 'app/actions';
import { getFilteredCommits } from 'app/selectors';

export class GitTemporal extends Component<
  GitTemporalProps & DispatchProps & StateProps
> {
  componentDidMount() {
    const { path, dispatch } = this.props;
    dispatch(selectPath(path));
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.path !== this.props.path) {
      const { dispatch, path } = nextProps;
      dispatch(selectPath(path));
    }
  }
  // handleRefreshClick = e => {
  //   e.preventDefault();

  //   const { dispatch, path } = this.props;
  //   dispatch(invalidatePath(path));
  //   dispatch(fetchCommitsIfNeeded(path));
  // };

  render() {
    const {
      selectedPath,
      serviceBaseUrl,
      isFetching,
      commits,
      authorNames = [],
    } = this.props;

    return (
      <div>
        {isFetching ? (
          <h2> Loading...</h2>
        ) : !commits || commits.length <= 0 ? (
          <h2>Empty.</h2>
        ) : (
          <div>
            <h1>I'm going to grow up to be a real app one day.</h1>
            <p>
              Unless Bee gets sick of TS causing her extra work and throws the
              baby out with the bath water.
            </p>
            <div>
              <label>Path:</label>
              {selectedPath}
            </div>
            <div>
              <label>baseServiceUrl:</label>
              {serviceBaseUrl}
            </div>
            <div>
              <label>number of commits:</label>
              {commits && commits.length}
            </div>
            <div>
              <label>{authorNames.length} Authors</label>
              {authorNames.map(authorName => (
                <div key={authorName}>{authorName}</div>
              ))}
            </div>
          </div>
        )}
      </div>
    );
  }
}
// debugger;
export function mapStateToProps(state) {
  return getFilteredCommits(state);
}

export default connect(mapStateToProps)(GitTemporal);
