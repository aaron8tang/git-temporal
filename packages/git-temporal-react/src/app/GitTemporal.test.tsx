import React from 'react';
import { shallow } from 'enzyme';
import { GitTemporal } from './GitTemporal';

const dispatchMock = jest.fn();

describe('components/GitTemporal', () => {
  test('it should match snapshot', () => {
    const wrapper = shallow(<GitTemporal dispatch={dispatchMock} />);
    expect(wrapper).toMatchSnapshot();
  });
});
