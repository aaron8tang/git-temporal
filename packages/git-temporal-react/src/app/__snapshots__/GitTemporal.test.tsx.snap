// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/GitTemporal when changing path prop it should match snapshot (should still be showing commits) 1`] = `
<div>
  <div>
    <h1>
      I'm going to grow up to be a real app one day.
    </h1>
    <p>
      Unless Bee gets sick of TS causing her extra work and throws the baby out with the bath water.
    </p>
    <div>
      <label>
        Path:
      </label>
    </div>
    <div>
      <label>
        baseServiceUrl:
      </label>
    </div>
    <div>
      <label>
        number of commits:
      </label>
      5
    </div>
    <div>
      <label>
        0
         Authors
      </label>
    </div>
  </div>
</div>
`;

exports[`components/GitTemporal when rendered with commits and isFetching it should match snapshot (it should show loading indicator) 1`] = `
<div>
  <h2>
     Loading...
  </h2>
</div>
`;

exports[`components/GitTemporal when rendered with commits and not fetching it should match snapshot (it should be showing commits) 1`] = `
<div>
  <div>
    <h1>
      I'm going to grow up to be a real app one day.
    </h1>
    <p>
      Unless Bee gets sick of TS causing her extra work and throws the baby out with the bath water.
    </p>
    <div>
      <label>
        Path:
      </label>
    </div>
    <div>
      <label>
        baseServiceUrl:
      </label>
    </div>
    <div>
      <label>
        number of commits:
      </label>
      5
    </div>
    <div>
      <label>
        0
         Authors
      </label>
    </div>
  </div>
</div>
`;

exports[`components/GitTemporal when rendered without props it should match snapshot (it should be showing empty indicator) 1`] = `
<div>
  <h2>
    Empty.
  </h2>
</div>
`;
